src/lazy/text/raw/sequence.rs

```diff
@@ -87,67 +88,42 @@ impl Debug for LazyRawTextList_1_0<'_> {
     has_returned_error: bool,
+    spooky: PhantomData<E>,
 }
 
-impl<'data> RawTextListIterator_1_0<'data> {
-    pub(crate) fn new(input: TextBuffer<'data>) -> RawTextListIterator_1_0<'data> {
-        RawTextListIterator_1_0 {
+impl<'data, E: TextEncoding<'data>> RawTextListIterator<'data, E> {
+    pub(crate) fn new(input: TextBuffer<'data>) -> RawTextListIterator<'data, E> {
+        RawTextListIterator {
             input,
             has_returned_error: false,
+            spooky: PhantomData,
         }
     }
 }
 
-impl RawTextListIterator_1_0<'_> {
-    pub(crate) fn find_span(&self) -> IonResult<Range<usize>> {
-        // The input has already skipped past the opening delimiter.
-        let start = self.input.offset() - 1;
-        // We need to find the input slice containing the closing delimiter. It's either...
-        let input_after_last = if let Some(value_result) = self.last() {
-            let value = value_result?.expect_value()?;
-            // ...the input slice that follows the last sequence value...
-            self.input
-                .slice_to_end(value.input.offset() + value.total_length() - self.input.offset())
-        } else {
-            // ...or there aren't values, so it's just the input after the opening delimiter.
-            self.input
-        };
-        let (mut input_after_ws, _ws) =
-            input_after_last
-                .match_optional_comments_and_whitespace()
-                .with_context("seeking the end of a list", input_after_last)?;
-        // Skip an optional comma and more whitespace
-        if input_after_ws.bytes().first() == Some(&b',') {
-            (input_after_ws, _) = input_after_ws
-                .slice_to_end(1)
-                .match_optional_comments_and_whitespace()
-                .with_context("skipping a list's trailing comma", input_after_ws)?;
-        }
-        let (input_after_end, _end_delimiter) = satisfy(|c| c == ']')(input_after_ws)
-            .with_context("seeking the closing delimiter of a list", input_after_ws)?;
-        let end = input_after_end.offset();
-        Ok(start..end)
-    }
-}
-
-impl<'data> Iterator for RawTextListIterator_1_0<'data> {
-    type Item = IonResult<LazyRawValueExpr<'data, TextEncoding_1_0>>;
+impl<'data, E: TextEncoding<'data>> Iterator for RawTextListIterator<'data, E> {
+    type Item = IonResult<LazyRawValueExpr<'data, E>>;
 
     fn next(&mut self) -> Option<Self::Item> {
         if self.has_returned_error {
             return None;
         }
-        match self.input.match_list_value() {
-            Ok((remaining, Some(value))) => {
-                self.input = remaining;
-                let value = RawValueExpr::ValueLiteral(LazyRawTextValue_1_0::from(value));
-                Some(Ok(value))
-            }
-            Ok((_remaining, None)) => {
+        let result = whitespace_and_then(alt((
+            "]".value(None),
```

Comment:
Should this one be `peek("]").value(None)` like the equivalent code for S-expressions?

Reply:
Yeah, I think so. Making it `peek` guarantees that you can keep calling `next()` on the iterator and it will still return `None`. Not critical, but a nice guard rail.